src/evmTest/resources/solidity/MinimumViableToken.sol

```diff
@@ -0,0 +1,19 @@
+pragma solidity ^0.4.9;
+
+contract MinimumViableToken {
+    /* This creates an array with all balances */
+    mapping (address => uint256) public balanceOf;
+
+    /* Initializes contract with initial supply tokens to the creator of the contract */
+    function MinimumViableToken(uint256 initialSupply) {
+        balanceOf[msg.sender] = initialSupply;              // Give the creator all initial tokens
+    }
+
+    /* Send coins */
+    function transfer(address _to, uint256 _value) {
+        if (balanceOf[msg.sender] < _value) throw;           // Check if the sender has enough
+        if (balanceOf[_to] + _value < balanceOf[_to]) throw; // Check for overflows
+        balanceOf[msg.sender] -= _value;                     // Subtract from the sender
+        balanceOf[_to] += _value;                            // Add the same to the recipient
```

Comment:
Regarding your first point
> I wonder what the balance updates compile to, because a single line balanceOf[_to] += _value; would suggest that you could possibly add value to the recipient's balance without deducting from sender's balance, which should not be allowed by EVM

balanceOf is just a map, you can update it however you want, can't you?